### ISSUE TITLE
Add a INV_PROJECTION_MATRIX built-in to Canvas Item Shader

### ISSUE THREAD

**Issue by ghsoares** [godotengine/godot-proposals#1195]
**Describe the project you are working on:**
A 2D project with repeating vaporwave pattern on floor

**Describe the problem or limitation you are having in your project:**
I'm creating a dynamic texture tiling with canvas item shader to be able to tile a texture seamless between objects using local pixel coordinates to world coordinates. The problem is the Matrix used to convert local space to world space is needed to be provided manually via script (with get_global_transform) on _ready and everytime the node moves. The workflow would be easier if this matrix is accessible directly in canvas item shader like INV_PROJECTION_MATRIX on spatial shader.

**Describe the feature / enhancement and how it helps to overcome the problem or limitation:**
This is simple as adding the INV_PROJECTION_MATRIX matrix to shading language, passing the node global transform automatically to the shader.

**Describe how your proposal will work, with code, pseudocode, mockups, and/or diagrams:**
Honestly I don't really know exactly how some built-ins is passed from CPU to GPU in shader, but it would work simply passing the node global transform to shader. Then the use would as simple as:

`vec2 world_position = (PROJECTION_MATRIX * vec4(VERTEX, 0.0, 1.0)).xy;`

**If this enhancement will not be used often, can it be worked around with a few lines of script?:**
Indeed yes, just assigning the global transform of the node with set_shader_param to shader.

**Is there a reason why this should be core and not an add-on in the asset library?:**
Because it makes the workflow easier to anyone create a more variety of shaders without depending on CPU part, as this is already automated.

**Comment by clayjohn** [comment]
Related: https://github.com/godotengine/godot/pull/24002

The consensus is that we would like to provide a full set of transforms for 2D the way we have for 3D. :)

**Comment by ghsoares** [comment]
@clayjohn indeed, is a very interesting feature that adds a large range of possibilities in Godot.

**Comment by QbieShay** [comment]
Hi, pitching in on this. I'm developing an interactive foliage system for 2D, and having a matrix where i can fetch the world origin from would help my usecase.
What i need it for is to have some wind in the scene, and i don't want all the bushes to sway together.

**Comment by QbieShay** [comment]
I wanted to add that the project that I'm working on should come out around Feb, so we can't quite wait for 4.0.
